src/hats_import/hipscat_conversion/run_conversion.py

```diff
@@ -0,0 +1,162 @@
+"""Convert a hipscatted catalog into a HATS catalog, with appropriate metadata/properties."""
+
+import json
+import tempfile
+
+import hipscat.pixel_math.healpix_shim as hp
+import numpy as np
+import pyarrow.parquet as pq
+from dask.distributed import as_completed, get_worker
+from dask.distributed import print as dask_print
+from hats.catalog import CatalogType, PartitionInfo, TableProperties
+from hats.io import file_io, parquet_metadata, paths
+
+import hats_import
+from hats_import.hipscat_conversion.arguments import ConversionArguments
+from hats_import.pipeline_resume_plan import print_progress
+
+
+def run(args, client):
+    """Run index creation pipeline."""
+    if not args:
+        raise TypeError("args is required and should be type ConversionArguments")
+    if not isinstance(args, ConversionArguments):
+        raise TypeError("args must be type ConversionArguments")
+
+    # Create basic properties, using catalog info, provenance info, and partition_info files
+    catalog_info = None
+    with (args.input_catalog_path / "catalog_info.json").open("r", encoding="utf-8") as json_file:
+        catalog_info = json.load(json_file)
+    provenance_info = None
+    with (args.input_catalog_path / "provenance_info.json").open("r", encoding="utf-8") as json_file:
+        provenance_info = json.load(json_file)
+
+    catalog_type = CatalogType(catalog_info["catalog_type"])
+    if catalog_type not in (
+        CatalogType.OBJECT,
+        CatalogType.SOURCE,
+        CatalogType.MARGIN,
+        CatalogType.ASSOCIATION,
+    ):
+        raise ValueError("Conversion only implemented for object, source, margin, and association tables")
+
+    catalog_info.pop("epoch", None)
+    if "tool_args" in provenance_info:
+        builder_str = (
+            provenance_info["tool_args"]["tool_name"]
+            + " v"
+            + provenance_info["tool_args"]["version"]
+            + " hats-importer conversion v"
+            + hats_import.__version__
+        )
+        catalog_info["hats_builder"] = builder_str
+        if "runtime_args" in provenance_info["tool_args"]:
```

Comment:
```suggestion
        if runtime_args := provenance_info["tool_args"].get("runtime_args"):
```
If we can count on Python 3.8 or later, this pattern is a nice way of combining this line and the next.